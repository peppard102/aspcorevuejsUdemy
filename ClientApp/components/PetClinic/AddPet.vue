<template>
    <div>
        <h1>Add New Pet</h1>
        <hr />
        <b-form @submit="onSubmit">
            <b-form-group id="input-group-2" label="Name:" label-for="input-2">
                <b-form-input id="input-2"
                              v-model="Item.name"
                              required
                              placeholder="Enter the pet's name"></b-form-input>
            </b-form-group>

            <b-button type="submit" variant="primary">Submit</b-button>
        </b-form>
        <br />
        <transition name="fade">
            <div class="alert alert-success" v-if="showAlert">
                You have successfully added the pet!
            </div>
        </transition>
    </div>
</template>

<script>
    export default {
        data() {
            return {
                Item: {},
                showAlert: false,
            }
        },
        methods: {
            onSubmit: function (evt) {
                evt.preventDefault()
                let self = this;

                try {
                    self.$http.post('/api/Pet', self.Item).then(result => {
                        self.showAlert = true;

                        setTimeout(function () {
                            self.showAlert = false;
                        }, 3000);

                        self.Item = {};
                    });
                    
                } catch (error) {
                    console.log(error)
                }
            }
        }
    }
</script>

<style>
    .fade-enter {
        opacity: 0;
    }

    .fade-enter-active {
        transition: opacity 2s;
    }

    .fade-leave-active {
        transition: opacity 2s;
        opacity: 0;
    }

    .alert-success {
        color: #155724;
        background-color: #d4edda;
        border-color: #c3e6cb;
    }
</style>
