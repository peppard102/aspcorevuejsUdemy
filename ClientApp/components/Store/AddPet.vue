<template>
    <div>
        <h1>{{msg}}</h1>
        <div class="form-horizontal">
            <fieldset>

                <!-- Form Name -->
                <legend>Add New Pet</legend>

                <!-- Text input-->
                <div class="form-group">
                    <label class="col-md-4 control-label" for="textinput">Name</label>
                    <div class="col-md-4">
                        <input v-model="Item.Name" placeholder="Mouse, Keyborad, etc." class="form-control input-md" type="text">

                    </div>
                </div>

                <!-- Button -->
                <div class="form-group">
                    <label class="col-md-4 control-label" for=""></label>
                    <div class="col-md-4">
                        <button @click="AddData()" class="btn btn-primary">Add</button>
                    </div>
                </div>


            </fieldset>
        </div>


    </div>
</template>

<script>
    export default {
        data() {
            return {
                msg: 'Add Pet',
                Item: {}
            }
        },
        methods: {
            AddData: function () {
                try {
                    this.$http.post('/api/Pet', this.Item).then(result => {
                        console.log(result);
                        alert(`Pet ${result.data.name} has been added successfully`);
                        this.Item = {};
                    });
                    
                } catch (error) {
                    console.log(error)
                }
            }
        }
    }
</script>

<style>

</style>
