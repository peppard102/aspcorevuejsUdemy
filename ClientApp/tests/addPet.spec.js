import { shallowMount } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import AddPet from '@/components/PetClinic/AddPet.vue'
import { createLocalVue } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'


// create an extended `Vue` constructor
const localVue = createLocalVue()

// install plugins as normal
localVue.use(BootstrapVue)

describe('AddPet.vue', () => {
    const wrapper = mount(AddPet, {
        localVue
    })

    it('Displays correct title', () => {
        expect(wrapper.text()).toMatch('Add New Pet')
    })

    it('has a button', () => {
        expect(wrapper.contains('button')).toBe(true)
    })

    test('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
})
