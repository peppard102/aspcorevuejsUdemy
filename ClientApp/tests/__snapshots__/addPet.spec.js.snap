// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPet.vue renders correctly 1`] = `
<div>
  <h1>
    Add New Pet
  </h1>
   
  <hr />
   
  <form
    class=""
  >
    <div
      class="form-group"
      id="input-group-2"
      role="group"
    >
      <label
        class="d-block"
        for="input-2"
        id="input-group-2__BV_label_"
      >
        Name:
      </label>
      <div>
        <input
          aria-required="true"
          class="form-control"
          id="input-2"
          placeholder="Enter the pet's name"
          required="required"
          type="text"
        />
        <!---->
        <!---->
        <!---->
      </div>
    </div>
     
    <button
      class="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>
   
  <br />
   
  <!---->
</div>
`;
